server/src/__tests__/users.spec.ts

```diff
@@ -0,0 +1,86 @@
+import 'reflect-metadata';
+import casual from 'casual';
+import { gql } from 'apollo-server-core';
+import { ApolloServer } from 'apollo-server';
+
+import { startServer } from '../server';
+import { UserModel } from '../models/users-model';
+import * as testConfig from '../config/test-config';
+
+const GET_USERS_BY_ROLE = gql`
+  query GetUsersByRole($role: String!) {
+    getUsersByRole(role: $role) {
+      lastname
+      firstname
+      mail
+      role
+    }
+  }
+`;
+const GET_USER = gql`
+  query GetUsersById($id: String!) {
+    getUserById(id: $id) {
+      id
+      lastname
+      firstname
+      mail
+    }
+  }
+`;
+
+const USER_MODEL_MOCK = () => ({
+  lastname: casual.last_name,
+  firstname: casual.first_name,
+  mail: casual.email,
+  birthday: casual.date(),
+  password: casual.password,
+  role: 'admin',
+  classrooms: [casual.uuid],
+});
+
+describe('src/resolvers/user-resolver =>', () => {
+  let apollo: ApolloServer;
+  beforeAll(async () => {
+    const serverConfig = await testConfig.initDatabase();
+    apollo = await startServer(serverConfig);
+
+    // insert multiple data in MongoMemorySever for test it
+    await UserModel.insertMany([
```

Comment:
not a big deal but I would move the fake users insertion into a separate function that I would call in the tests
becaus one may want to test with an empty db

Reply:
yes indeed, but for now we don't have test for it. I will export this function for next step